src/hotspot/share/memory/virtualspace.hpp

```diff
@@ -58,25 +59,29 @@ class ReservedSpace {
   //                       0 during initialization.
   void clear_members();
   void initialize_members(char* base, size_t size, size_t alignment,
-                          size_t page_size, bool special, bool executable);
+                          size_t page_size, bool special, bool executable, MEMFLAGS flag);
 
   void initialize(size_t size, size_t alignment, size_t page_size,
                   char* requested_address, bool executable);
 
   void reserve(size_t size, size_t alignment, size_t page_size,
                char* requested_address, bool executable);
  public:
+
+  MEMFLAGS nmt_flag() { return _nmt_flag; }
+  void set_nmt_flag(MEMFLAGS flag) { _nmt_flag = flag; }
```

Comment:
I have a feeling that set_nmt_flag should not exist and be replaced by updated initialize functions.